src/transactions/transactionBuilder/transactionBuilder.ts

```diff
@@ -387,12 +383,7 @@ export function generateSignedTransactionForSimulation(args: InputSimulateTransa
 export function getAuthenticatorForSimulation(publicKey: PublicKey) {
   // TODO add support for AnyMultiKey
   if (publicKey instanceof AnyPublicKey) {
-    if (publicKey.publicKey instanceof Ed25519PublicKey) {
-      return new AccountAuthenticatorSingleKey(publicKey, new AnySignature(new Ed25519Signature(new Uint8Array(64))));
-    }
-    if (publicKey.publicKey instanceof Secp256k1PublicKey) {
-      return new AccountAuthenticatorSingleKey(publicKey, new AnySignature(new Secp256k1Signature(new Uint8Array(64))));
-    }
+    return new AccountAuthenticatorSingleKey(publicKey, new AnySignature(new Ed25519Signature(new Uint8Array(64))));
```

Comment:
Is this because it's a hack around Secp256k1 and the signature not being checked properly?

Perhaps we can just leave instance of Secp, and then default to Ed25519

Reply:
Technically not a hack, in rust you can verify any signature against a publicKey.
I think the code would be much cleaner this way, maybe I can add a comment?